### PR TITLE
Made an Actual Website

### PR DESCRIPTION
**Pull request by Thanson16**
Hello everybody,

In vein of issue #1, I used Hugo and some screenshots I took to make an actual website. It isn't perfect but I think it is ready enough to be put to use. You can see a hosted version of the result here:
https://thanson16.github.io/piqueserver.github.io/

If it meets your standards, feel free to merge it in. If you do, I will (in the near future) fix up a couple of the outstanding problems:
- Add links to the documentation
- Fix the missing icon for Gitter
- Add a readme to explain how it works and how to update its content

Hopefully I'm not missing anything...I just put this together from about 2am to 7am local time, and very much should get some sleep now. I look forward to hearing your feedback.

### CODE REVIEW

**Comment by NotAFile** [comment]
Nice! I like it.

I'd like to see a change to the colors though. Piqueserver's logo color is a blue, and we don't really use orange anywhere.

I'll have a closer look at this soon, and maybe make some changes, please remind me if I forget.

**Comment by Thanson16** [comment]
Ah, color! Orange was the default but I shall see if I can change it to blue. Good idea!

**Comment by Thanson16** [comment]
Should have all instances of oranges changed to blues to match the icon now. Can preview here again:
https://thanson16.github.io/piqueserver.github.io/

**Comment by samuelallan72** [comment]
@Thanson16 Nice! This is amazing! :smile:

**Comment by samuelallan72** [comment]
One question, can GitHub or a free CI build the site for us to avoid needing the output files stored in the repository?

**Comment by Thanson16** [comment]
It would be pretty straitforward to set up a CI to do the building and then push it to a gh-pages branch (have done this using Jenkins at work). Do you guys currently have any kind of a CI?

**Comment by samuelallan72** [comment]
We're connected to Travis and Appveyor for https://github.com/piqueserver/piqueserver . Travis should be able to do it I would imagine.

**Comment by Thanson16** [comment]
Yeah that should work. Maybe I'll look into writing a file for that purpose, although I'm not sure how much of that I'll be able to do without access.

Usual practice is to either have a separate branch hold the "built" files or to have a specific folder in the repo do it; I'd probably go with the branch (gh-pages) method because it'd keep it nice and clean.

To build the site, run `hugo -D` in the repo root. This will generate the site and place it in the `/public` folder. You'll need Hugo installed on the build machine.

**Comment by samuelallan72** [comment]
@Thanson16 I connected [travis](https://travis-ci.org/piqueserver/piqueserver.github.io/) to this repo, so it should start building the branch once there's a .travis.yml file in the root. Maybe start with getting the site built through travis, and then a project member can help set up the config in travis necessary to push to the gh-pages branch.

**Comment by Thanson16** [comment]
Okay, cool! Will it build this even though it is still just a pull request?

**Comment by samuelallan72** [comment]
@Thanson16 Yes. Not sure how to get the artefacts out to check the results though ¯\\\_(ツ)\_/¯

**Comment by Thanson16** [comment]
@swalladge
Okay, so here's where it's at.

I cleaned up the icon situation a little, added a link to the documentation, and adjusted the background image. I then removed all the unneeded build stuff from the top level.

Next, I added a `.travis.yml`. As it is now, it seems to run and build the website and gives no errors. The following would need to be done to see if it can successfully push to Pages:

- API key with correct access needs to be added into Github
- That API key needs to be added into Travis with name `$PAGES_GITHUB_TOKEN`
- gh-pages branch needs to be created (contents don't matter)
- Pull request needs to be merged into a branch (deploy step is set to run on all branches currently, but I cannot get it to run deploy on a pull request). Doesn't have to be master; any branch should work.

If that successfully pushes to gh-pages, then the next steps would be:
- Edit `.travis.yml` to only deploy on master
- Change the GitHub Pages settings to source from the gh-pages branch
- Merge intermediate branch into master.

Let me know if I'm missing something or if there is something else you'd like me to do in the meantime.

Thank you all for your hard work; this is an awesome project and I really appreciate it.

**Comment by samuelallan72** [comment]
@Thanson16 Wow, awesome! Thanks for your work on this! :)

I think it would be fine to do whatever tinkering is required to deploy once it's got the general approval and merged to master. I'll have a poke through the files now to review. :+1:

File: config.toml
Comment (samuelallan72) [inline config.toml]: baseurl should be left set to `""` to aid local development and make the site more portable.

File: public/categories/index.xml
Comment (samuelallan72) [inline public/categories/index.xml]: All files in the `public` directory should be removed from git.

File: config.toml
Comment (samuelallan72) [inline config.toml]: tabs and spaces are mixed in this file. Please normalize to spaces. :)

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
@NotAFile should the `latest_version.json` file be accessible via www.piqueserver.org? @Thanson16 if so, this file should be copied into the public dir as part of the build process.

Is the theme in here a copy of https://github.com/digitalcraftsman/hugo-creative-theme ? If it is, or if it is with few modifications, it would be nice to include it as a git submodule rather than copying the contents. That would allow us to easily update if the theme updates, and easier to track changes in this repo.

Once all the spurious files are removed, it would be nice to squash the changes so we don't have an unnecessarily large git history.

Also please feel free to sing out if you'd like a hand making any of these changes for any reason, or if you have questions/comments/arguments. :smile:

File: config.toml
Comment (Thanson16) [reply config.toml]: Would this get set when Travis builds then? Or will Hugo be fine with it being empty?

File: config.toml
Comment (feikname) [inline config.toml]: piqueserver is already migrated to Python 3. Also, Pysnip should be written as PySnip.

File: config.toml
Comment (samuelallan72) [reply config.toml]: I think it will be fine (all links will end up being `/path/to/file` instead of `https://www.piqueserver.org/path/to/file`. Not sure how to handle generated urls that *should* be absolute though, like canonical metadata urls or rss feed urls. I guess we can worry later.

Comment (samuelallan72) [reply config.toml]: Although, not sure how it works through your test github.io domain. :thinking:

File: config.toml
Comment (feikname) [inline config.toml:44]: The link here is kinda hard to read on the website, not sure how it could be fixed though.

Perhaps a brighter color such as `#00bbe3` (and a lighter color on hover also, accordingly)?

File: config.toml
Comment (Thanson16) [reply config.toml]: My shower thoughts this morning thought that leaving it blank should work in all scenarios that this version of the website would deal with, but, like you say, stuff like RSS feeds would probably get broken. That'd probably be something we could have Travis edit before running Hugo if it was an issue down the road.

File: themes/hugo-creative-theme/layouts/index.html
Comment (feikname) [inline themes/hugo-creative-theme/layouts/index.html]: The navigation links are also a bit difficult to read / spot in this background. Reverting the opacity from 0.7 to 1.0 (making it plain white) should suffice to fix it. Conversely, the hover color should be changed to use 0.7 opacity instead of being plain white.

**Comment by Thanson16** [comment]
@swalladge  I resolved the three marked items, which just leaves the two things mentioned in your paragraph.

I'll wait and see what @NotAFile says on the `latest_version.json`.

As for the template, yes, it is a copy of it. I sliced out all of its git stuff because I had to make some changes to the template itself in a way that I wasn't sure future updates would be useful, but perhaps that was the wrong move. Regardless, submodules are not something I have experience with, so I'd probably be reading up on this tomorrow so I can fix it if it is still something you want.

File: config.toml
Comment (feikname) [inline config.toml]: please remove everything since the `?` in the link (the question mark should also be deleted).

File: themes/hugo-creative-theme/layouts/partials/contact.html
Comment (feikname) [inline themes/hugo-creative-theme/layouts/partials/contact.html]: there should be a space after `if .Site.Params.contact.documentation` and before `}}` here for consistency.

File: themes/hugo-creative-theme/layouts/partials/contact.html
Comment (feikname) [inline themes/hugo-creative-theme/layouts/partials/contact.html]: I don't know the syntax, should an `and` right after the `if` really be here?

File: themes/hugo-creative-theme/layouts/partials/contact.html
Comment (feikname) [inline themes/hugo-creative-theme/layouts/partials/contact.html]: Shouldn't here be `.Site.Params.contact.matrix` instead?

**Comment by samuelallan72** [comment]
@Thanson16 

> As for the template, yes, it is a copy of it. 

In that case, assuming we can work with the vanilla upstream theme, a submodule would be good please. It will help keep the repo as clean as possible. I'd be happy to link it in as a submodule if you'd prefer (or read up on it if you'd like to do it yourself; submodules are always a handy topic to know about).

**Comment by Thanson16** [comment]
@swalladge If you're good with linking it in, that'd be appreciated, as long as we are still able to apply those couple of modifications I made. I'll be looking through the other comments and fixing those items in the meantime; I should have time to look at all them this weekend.

File: themes/hugo-creative-theme/layouts/partials/contact.html
Comment (Thanson16) [reply themes/hugo-creative-theme/layouts/partials/contact.html]: This was in the original template; after examining it it seems to be correct...looks like the "and" applies to the two parts following it.

File: themes/hugo-creative-theme/layouts/partials/contact.html
Comment (Thanson16) [reply themes/hugo-creative-theme/layouts/partials/contact.html]: Fixed

File: themes/hugo-creative-theme/layouts/partials/contact.html
Comment (Thanson16) [reply themes/hugo-creative-theme/layouts/partials/contact.html]: Fixed

File: config.toml
Comment (Thanson16) [reply config.toml]: Fixed

File: themes/hugo-creative-theme/layouts/index.html
Comment (Thanson16) [reply themes/hugo-creative-theme/layouts/index.html]: I have improved this, not quite in the way you suggested. I darkened the background image further, then made the background slightly dark as well. At the suggestion of a friend, I also bumped up the font size. You can see a preview of it here, as before:

https://thanson16.github.io/piqueserver.github.io/

Let me know if you think it is good enough...I personally think it looks good now, but I'm open to changing it more.

**Comment by samuelallan72** [comment]
@Thanson16 I converted the theme to a submodule, and moved the files you had edited to the appropriate locations in the repository for hugo to find and use as overrides. Sorry, I think this means your last few commits will be lost. :/

Please pull and run `git submodule update` to pull the submodule.

File: config.toml
Comment (Thanson16) [reply config.toml]: fixed

**Comment by Thanson16** [comment]
@swalladge Darn haha. I'll have to do that over; np. Thanks for doing that!

File: config.toml
Comment (Thanson16) [reply config.toml:44]: Brightened it some, but not quite that blue (trying to keep it more in tone with the existing palette).

**Comment by Thanson16** [comment]
@swalladge @feikname , I think all the fixes and concerns discussed are now resolved, assuming @swalladge is good with the couple of aesthetic fixes I made for readability.

A copy of the current state of it can be previewed here:
https://thanson16.github.io/piqueserver.github.io/

Let me know if anything else concerns you guys or if I missed something after the submodule conversion.

File: themes/hugo-creative-theme/layouts/index.html
Comment (feikname) [reply themes/hugo-creative-theme/layouts/index.html]: It's quite readable now, thanks!

**Comment by feikname** [comment]
Unfortunately GitHub review doesn't quite work with submodule files, so I will post it as a normal comment.


In [line 15 of `themes/hugo-creative-theme/static/js/creative.js`](
https://github.com/digitalcraftsman/hugo-creative-theme/blob/cea31c1813c13e76764c84417de3578b2ad876ed/static/js/creative.js#L10-L17) there's this:

```js
        }, 1250, 'easeInOutExpo');
```

Upon testing, to me it feels like the animation for clicking in the header links takes too long and that the selected [easing](https://easings.net/en) is too abrupt for shorter values.

A good compromise I found was that using the following instead yields a faster and smoother experience overall:

```js
        }, 750, 'easeInOutSine');
```

Could you please change that file to that? Or at least something quicker and smoother than the current.

**Review by feikname — APPROVED** [review]
Besides the last change I requested, ~~LGTM~~ it seems everything is fine and ready to merge after a bit of commit cleanup 👍 

@NotAFile 
> I'll have a closer look at this soon, and maybe make some changes, please remind me if I forget.

Pinging in case you forgot it 😛

**Comment by samuelallan72** [comment]
quick note from my phone: we don't need to change anything in the submodule - use the override file in the root directly tree (I think we already have a creative.css file overridden in the repo to use)

November 19, 2019 11:26:29 AM UTC "Flávio Monteiro" <notifications@github.com> kirjoitti:
>Unfortunately GitHub review doesn't quite work with submodule files, so
>I will post it as a normal comment.
>
>
>In [line 15 of `themes/hugo-creative-theme/static/js/creative.js`](
>https://github.com/digitalcraftsman/hugo-creative-theme/blob/cea31c1813c13e76764c84417de3578b2ad876ed/static/js/creative.js#L10-L17)
>there's this:
>
>```js
>        }, 1250, 'easeInOutExpo');
>```
>
>Upon testing, to me it feels like the animation for clicking in the
>header links takes too long and that the selected
>[easing](https://easings.net/en) is too abrupt for shorter values.
>
>A good compromise I found was that using the following instead yields a
>faster and smoother experience overall:
>
>```js
>        }, 750, 'easeInOutSine');
>```
>
>Could you please change that file to that? Or at least something
>quicker and smoother than the current.

-- 
Envoyé de mon appareil Android avec Courriel K-9 Mail. Veuillez excuser ma brièveté.

**Comment by Thanson16** [comment]
@feikname  Changed the animation as you requested. Note that I did not update the version being hosted on my Pages.

**Comment by NotAFile** [comment]
One more thing I've noticed is that the contrast between the text and background image is very low. Firefox accessibility inspector confirms this (contrast ratio 1.6-4.5 out of the suggested). The best thing here is to probably do something with the background image, like blur it, or use one with more contrast.

**Comment by Thanson16** [comment]
@NotAFile How's this? I added a darker image. Reanalyzing with Firefox showed improvement (scoring in the low 3.x on the low end instead of 1.x before).
https://thanson16.github.io/piqueserver.github.io/
You may need to clear your cache for the new image to load.

**Comment by Thanson16** [comment]
@NotAFile  I also made the homepage fonts whiter, which made Firefox's accessibility tool deem them acceptable. Preview has been pushed as well. I'm thinking it is probably good now, but maybe you'd like to see something else.

**Review by NotAFile — APPROVED** [review]
I'm super happy with this.

Thanks so much for your work!

**Comment by NotAFile** [comment]
@swalladge the versions.json file was, afaik never used. Our implementation of version check uses the github API instead.

I'm going to let you do the final merge in case you have any last minute suggestions.

**Comment by Thanson16** [comment]
@NotAFile , glad to hear you like it!

Thanks all of you for your hard work on piqueserver. It is really awesome, and I'm glad I'm able to do something to be of use. You've been very kind and easy to work with.

File: static/css/all.css
Comment (samuelallan72) [inline static/css/all.css]: @Thanson16 can this be renamed to be clear that it's font-awesome?

**Review by samuelallan72 — APPROVED** [review]
@Thanson16 Awesome work! One small nitpick re the all.css naming and this is ready to merge.

Would you mind squashing all the changes in this branch to a single commit please? Normally it would be fine, but it's not a fun history with all the build artefacts and theme files being added and removed.

When it's squashed, I'll verify that the changes are the same as the previously unsquashed changes, and then merge.

File: static/css/all.css
Comment (Thanson16) [reply static/css/all.css]: fixed

**Comment by Thanson16** [comment]
@swalladge  I hope I didn't just break anything - this is my first time squashing from the command line (I usually use the button when I merge). If it worked, this should include everything. If I need to fix something let me know.

**Comment by samuelallan72** [comment]
@Thanson16 Looks good to me. :+1: Usually when squashing you should write up a commit message that describes the changes in that new squashed commit as a whole, rather than keeping the log of changes, but that's fine for now.

**Comment by samuelallan72** [comment]
@Thanson16 CC @NotAFile @feikname I sorted out the deploy config with travis. Needed to move the main source branch to something other than master (github forces pages to serve from master). New main branch is `develop`. I also created @piqueserver-travis user to use a more tightly scoped personal access token for travis to use to deploy the site.

TL;DR: https://piqueserver.org :rocket: :tada:

**Comment by Thanson16** [comment]
@swalladge Awesome!

Just so you know, GitHub Pages will let you serve from a branch titled `gh-pages`. However, in order to do so I think you need to first create the branch (otherwise the settings won't let you select it). But if this works and you're good with it, then great!

**Comment by samuelallan72** [comment]
@Thanson16 apparently not for organization pages repos :/

**Comment by Thanson16** [comment]
@swalladge Hmm....TIL. An odd limitation.

**Pull request merged**
